instrumentation/mongo/mongo-3.1/src/test/groovy/MongoClientTest.groovy

```diff
@@ -68,6 +68,27 @@ class MongoClientTest extends MongoBaseTest {
     collectionName = "testCollection"
   }
 
+  def "test create collection with already built ClientSettings"() {
```

Comment:
I don't quite understand how this test is related to the change in `MongoClientAdvice`. I would expect that you test e.g. calling `build` method twice and then check that only one span is created. Do I miss something?

Reply:
@iNikem it's using an existing `client.mongoClientOptions` (which will already have our `CommandListener`), so it tests that we don't add it again on top of that

Reply:
Why `client.mongoClientOptions` will already have a `CommandListener`? And in any case this then should be documented in test :) I always assume that if some code was confusing to me, it will probably be confusing to somebody else :)

Reply:
👍 comment here would be great

Reply:
Thanks for the comment. I completely agree. Initially, I was also little not sure about readability of this test. I added a comment now. Let me know if that's alright.